From cb2577f84ea2c8a884cdd5a1d7d87c80438db1bb Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Fri, 9 Aug 2013 14:32:30 +0800
Subject: [PATCH 463/635] ENGR00274593 Build failed with -Werror in android
 gcc toolchain

Fix compiling failure with -Werror in android gcc toolchain

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 mx5x/hwcomposer/hwcomposer.cpp       | 2 +-
 mx5x/libcamera/CameraModule.cpp      | 3 ++-
 mx5x/libcamera/V4l2CapDeviceBase.cpp | 4 ++--
 mx5x/libcamera/V4l2UVCDevice.cpp     | 4 ++--
 mx5x/libgralloc/gralloc.cpp          | 3 ++-
 mx6/libcamera2/Android.mk            | 2 +-
 mx6/libcamera2/CameraModule.cpp      | 2 ++
 7 files changed, 12 insertions(+), 8 deletions(-)

diff --git a/mx5x/hwcomposer/hwcomposer.cpp b/mx5x/hwcomposer/hwcomposer.cpp
index c401a70..bd4f594 100755
--- a/mx5x/hwcomposer/hwcomposer.cpp
+++ b/mx5x/hwcomposer/hwcomposer.cpp
@@ -60,7 +60,7 @@ hwc_module_t HAL_MODULE_INFO_SYM = {
 
 /*****************************************************************************/
 
-static void dump_layer(hwc_layer_t const* l) {
+static void dump_layer(hwc_layer_1_t const* l) {
     ALOGD("\ttype=%d, flags=%08x, handle=%p, tr=%02x, blend=%04x, {%d,%d,%d,%d}, {%d,%d,%d,%d}",
             l->compositionType, l->flags, l->handle, l->transform, l->blending,
             l->sourceCrop.left,
diff --git a/mx5x/libcamera/CameraModule.cpp b/mx5x/libcamera/CameraModule.cpp
index d28a3d3..bd9cf3c 100644
--- a/mx5x/libcamera/CameraModule.cpp
+++ b/mx5x/libcamera/CameraModule.cpp
@@ -15,7 +15,7 @@
  */
 
 /*
- * Copyright 2009-2012 Freescale Semiconductor, Inc.
+ * Copyright 2009-2013 Freescale Semiconductor, Inc.
  */
 
 #define LOG_TAG "CameraHAL"
@@ -61,6 +61,7 @@ camera_module_t HAL_MODULE_INFO_SYM = {
     },
     get_number_of_cameras: camera_get_number_of_cameras,
     get_camera_info: camera_get_camera_info,
+    set_callbacks: NULL,
 };
 
 typedef struct fsl_camera_device {
diff --git a/mx5x/libcamera/V4l2CapDeviceBase.cpp b/mx5x/libcamera/V4l2CapDeviceBase.cpp
index 9d1bcee..77d82dd 100755
--- a/mx5x/libcamera/V4l2CapDeviceBase.cpp
+++ b/mx5x/libcamera/V4l2CapDeviceBase.cpp
@@ -15,7 +15,7 @@
  */
 
 /*
- * Copyright 2009-2012 Freescale Semiconductor, Inc.
+ * Copyright 2009-2013 Freescale Semiconductor, Inc.
  */
 #include <string.h>
 #include <unistd.h>
@@ -690,7 +690,7 @@ namespace android{
         }
 
         for (unsigned int i = 0; i < mBufQueNum; i++) {
-            if (mCaptureBuffers[i].length && (mCaptureBuffers[i].virt_start > 0)) {
+            if (mCaptureBuffers[i].length && (mCaptureBuffers[i].virt_start != NULL)) {
                 munmap(mCaptureBuffers[i].virt_start, mCaptureBuffers[i].length);
                 mCaptureBuffers[i].length = 0;
                 CAMERA_LOG_RUNTIME("munmap buffers 0x%x\n", (unsigned int)(mCaptureBuffers[i].virt_start));
diff --git a/mx5x/libcamera/V4l2UVCDevice.cpp b/mx5x/libcamera/V4l2UVCDevice.cpp
index 5e198de..9c9e7df 100644
--- a/mx5x/libcamera/V4l2UVCDevice.cpp
+++ b/mx5x/libcamera/V4l2UVCDevice.cpp
@@ -15,7 +15,7 @@
  */
 
 /*
- * Copyright 2009-2012 Freescale Semiconductor, Inc.
+ * Copyright 2009-2013 Freescale Semiconductor, Inc.
  */
 
 #include <string.h>
@@ -312,7 +312,7 @@ CAPTURE_DEVICE_RET V4l2UVCDevice::V4l2DeAlloc()
     }
 
     for (unsigned int i = 0; i < mBufQueNum; i++) {
-        if (mUvcBuffers[i].length && (mUvcBuffers[i].virt_start > 0)) {
+        if (mUvcBuffers[i].length && (mUvcBuffers[i].virt_start != NULL)) {
             munmap(mUvcBuffers[i].virt_start, mUvcBuffers[i].length);
             mUvcBuffers[i].length = 0;
             CAMERA_LOG_RUNTIME("munmap buffers 0x%x\n", (unsigned int)(mUvcBuffers[i].virt_start));
diff --git a/mx5x/libgralloc/gralloc.cpp b/mx5x/libgralloc/gralloc.cpp
index 1e77b79..991c7f6 100644
--- a/mx5x/libgralloc/gralloc.cpp
+++ b/mx5x/libgralloc/gralloc.cpp
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-/* Copyright 2009-2012 Freescale Semiconductor, Inc. */
+/* Copyright 2009-2013 Freescale Semiconductor, Inc. */
 
 #include <limits.h>
 #include <unistd.h>
@@ -100,6 +100,7 @@ struct private_module_t HAL_MODULE_INFO_SYM = {
         lock: gralloc_lock,
         unlock: gralloc_unlock,
         perform: gralloc_perform,
+        lock_ycbcr: 0,
         reserved_proc: {0},
     },
     framebuffer: 0,
diff --git a/mx6/libcamera2/Android.mk b/mx6/libcamera2/Android.mk
index 199da4a..d99e9d1 100755
--- a/mx6/libcamera2/Android.mk
+++ b/mx6/libcamera2/Android.mk
@@ -83,7 +83,7 @@ endif
 #Define this for switch the Camera through V4L2 MXC IOCTL
 #LOCAL_CPPFLAGS += -DV4L2_CAMERA_SWITCH
 
-#LOCAL_CPPFLAGS += -Werror
+LOCAL_CPPFLAGS += -Werror
 
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
 LOCAL_MODULE:= camera.$(TARGET_BOARD_PLATFORM)
diff --git a/mx6/libcamera2/CameraModule.cpp b/mx6/libcamera2/CameraModule.cpp
index 868e70d..70af149 100755
--- a/mx6/libcamera2/CameraModule.cpp
+++ b/mx6/libcamera2/CameraModule.cpp
@@ -58,6 +58,7 @@ camera_module_t HAL_MODULE_INFO_SYM = {
     },
     get_number_of_cameras: camera_get_number_of_cameras,
     get_camera_info: camera_get_camera_info,
+    set_callbacks: NULL,
 };
 
 typedef struct fsl_camera_device {
@@ -278,6 +279,7 @@ camera2_device_ops_t fsl_camera_ops = {
     set_notify_callback:                 set_notify_callback,
     get_metadata_vendor_tag_ops:         get_metadata_vendor_tag_ops,
     dump:                                camera_dump,
+    get_instance_metadata:               NULL,
 };
 
 int camera_device_close(hw_device_t *device)
-- 
1.8.0

